conans/server/conf/__init__.py

```diff
@@ -80,51 +82,56 @@ def _get_file_conf(self, section, varname=None):
 
     @property
     def ssl_enabled(self):
-        if self.env_config["ssl_enabled"]:
-            return self.env_config["ssl_enabled"] == "true" or \
-                   self.env_config["ssl_enabled"] == "1"
-        else:
-            return self._get_file_conf("server", "ssl_enabled").lower() == "true" or \
-                   self._get_file_conf("server", "ssl_enabled").lower() == "1"
+        try:
+            ssl_enabled = self._get_conf_server_string("ssl_enabled").lower()
+            return ssl_enabled == "true" or \
+                   ssl_enabled == "1"
+        except ConanException:
+            return None
 
     @property
     def port(self):
-        if self.env_config["port"]:
-            return int(self.env_config["port"])
-        else:
-            return int(self._get_file_conf("server", "port"))
+        return int(self._get_conf_server_string("port"))
 
     @property
     def public_port(self):
-        if self.env_config["public_port"]:
-            return int(self.env_config["public_port"])
-        elif self._get_file_conf("server", "public_port"):
-            return int(self._get_file_conf("server", "public_port"))
-        else:
+        try:
+            return int(self._get_conf_server_string("public_port"))
+        except ConanException:
             return self.port
 
     @property
     def host_name(self):
-        return self._get_conf_server_string("host_name")
+        try:
+            return self._get_conf_server_string("host_name")
+        except ConanException:
+            return None
 
     @property
     def public_url(self):
-        protocol = "https" if self.ssl_enabled else "http"
-        port = ":%s" % self.public_port if self.public_port != 80 else ""
-        return "%s://%s%s/v1" % (protocol, self.host_name, port)
+        host_name = self.host_name
+        ssl_enabled = self.ssl_enabled
+        protocol_version = "v1"
+        if host_name is None and ssl_enabled is None:
```

Comment:
While I agree that the logic is correct, maybe a new conf parameter to explicitly enable the new behavior would be desired, if something could break (e.g. sysadmins upgrading the server but not the clients)

Reply:
I did not introduce a new setting since I wan't to reduce the required options and not increase it. I think the risk of breaking existing installations is pretty low given that every currently used conan_server necessarily has host_name and ssl_enabled defined. Upgrading the server, without manually removing these options from the existing config, therefore does not change the behavior in any way.

Reply:
Yes, that is true, only local servers could have those to None. Thanks!